cpp/main.cpp

```diff
@@ -107,11 +111,12 @@ int main()
 
         taggedPostCount[i] = 0;
 
-        uint8_t top5[5] = {0, 0, 0, 0, 0};
+        uint8_t top5[TOPN] = {0, 0, 0, 0, 0};
+        int related[TOPN] = {0, 0, 0, 0, 0};
         uint8_t minTags = 0;
 
         //  custom priority queue to find top N
-        for (int j = 0; j < taggedPostCount.size(); j++)
+        for (size_t j = 0; j < total; j++)
```

Comment:
isn't this bound-check evading?

Reply:
How so? In the logic of the code we know the length of the posts file after reading no it is the same length for this array.

Reply:
storing the size of the input at runtime is allowed.